### PR TITLE
rlp_derive: cleanup

### PR DESCRIPTION
**Pull request by ordian**
update deps and 2018 edition (in preparation for create/create2 fixup)

### CODE REVIEW

**Comment by debris** [comment]
why is this pr marked `patchstable`? it's insignificant and doesn't fix anything

**Comment by ordian** [comment]
> why is this pr marked `patchstable`? it's insignificant and doesn't fix anything

:point_down: 
> (in preparation for create/create2 fixup)

I'll submit a PR based on this for the issue https://github.com/paritytech/parity-ethereum/pull/11427 was trying to fix, but it was not a proper fix, since it didn't work if the db was synced with 2.7.0

File: util/rlp-derive/src/de.rs [hunk 1]
```diff
@@ -62,8 +66,7 @@ pub fn impl_decodable(ast: &syn::DeriveInput) -> TokenStream {
 	};
 
 	quote! {
-		#[allow(non_upper_case_globals, unused_attributes, unused_qualifications)]
-		const #dummy_const: () = {
+		const _: () = {
 			extern crate rlp;
```
Comment (niklasad1) [inline util/rlp-derive/src/de.rs]: I guess that this can be removed.

Comment (ordian) [reply util/rlp-derive/src/de.rs]: although `rlp_derive` crate uses 2018 edition, we cannot make this assumption about consumer crates (and it's actually not true)

Comment (niklasad1) [reply util/rlp-derive/src/de.rs]: ok, I see

**Pull request merged**